File: OrchidCore/src/main/java/com/eden/orchid/impl/server/admin/lists/CompilersList.java
package com.eden.orchid.impl.server.admin.lists;

import com.eden.orchid.api.compilers.OrchidCompiler;
import com.eden.orchid.api.server.admin.AdminList;

import javax.inject.Inject;
import java.util.Collection;
import java.util.Set;
import java.util.TreeSet;

public final class CompilersList implements AdminList<OrchidCompiler> {

    private Set<OrchidCompiler> list;

    @Inject
    public CompilersList(Set<OrchidCompiler> list) {
        this.list = new TreeSet<>(list);
    }

    @Override
    public String getKey() {
        return "compilers";
    }

    @Override
    public Collection<OrchidCompiler> getItems() {
        return list;
    }

    @Override
    public String getItemId(OrchidCompiler item) {
        return item.getClass().getSimpleName();
    }

    @Override
    public OrchidCompiler getItem(String id) {
        for(OrchidCompiler item : getItems()) {
            if(id.equals(item.getClass().getSimpleName())) {
                return item;
            }
        }
        return null;
    }
}

File: plugins/OrchidPluginDocs/src/main/kotlin/com/eden/orchid/pluginDocs/lists/CompilersList.kt
package com.eden.orchid.pluginDocs.lists

import com.eden.orchid.api.compilers.OrchidCompiler
import com.eden.orchid.api.server.admin.AdminList
import java.util.TreeSet
import javax.inject.Inject

@JvmSuppressWildcards
class CompilersList @Inject
constructor(list: Set<OrchidCompiler>) : AdminList {

    private val list: Set<OrchidCompiler>

    init {
        this.list = TreeSet(list)
    }

    override fun getKey(): String {
        return "compilers"
    }

    override fun getItems(): Collection<Any> {
        return list
    }

    override fun getItemId(item: Any): String {
        return item.javaClass.simpleName
    }

    override fun getItem(id: String): Any? {
        for (item in items) {
            if (id == item.javaClass.simpleName) {
                return item
            }
        }
        return null
    }
}

